src/main/java/it/pagopa/pn/service/desk/service/impl/ProfileServiceImpl.java

```diff
@@ -0,0 +1,100 @@
+                                userAttributesClient.getCourtesyAddressBySender(internalId, SENDER_ID_DEFAULT)
+                                        .collectList()
+                                        .onErrorResume(exception -> {
+                                            log.error("errorReason = {}, error = {}, Error during getCourtesyAddressBySender", exception.getMessage(), exception);
+                                            return Mono.error(new PnGenericException(ERROR_ON_USER_ATTRIBUTES_CLIENT, exception.getMessage()));
+                                        })
+                                        .map(courtesyDigitalAddressDtos -> ProfileMapper.getAddress(legalDigitalAddressDtos, courtesyDigitalAddressDtos, response))
+                                        .map(profileResponse -> internalId)
+                        )
+                )
+                .flatMap(internalId -> mandateClient.listMandatesByDelegator(internalId)
+                        .collectList()
+                        .onErrorResume(exception -> {
+                            log.error("errorReason = {}, error = {}, Error during listMandatesByDelegator", exception.getMessage(), exception);
+                            return Mono.error(new PnGenericException(ERROR_ON_MANDATE_CLIENT, exception.getMessage()));
+                        })
+                        .flatMap(internalMandateDelegators ->
+                                mandateClient.listMandatesByDelegate(internalId)
+                                        .collectList()
+                                        .onErrorResume(exception -> {
+                                            log.error("errorReason = {}, error = {}, Error during listMandatesByDelegate", exception.getMessage(), exception);
+                                            return Mono.error(new PnGenericException(ERROR_ON_MANDATE_CLIENT, exception.getMessage()));
+                                        })
+                                        .map(internalMandateDelegates -> ProfileMapper.getMandate(internalMandateDelegators, internalMandateDelegates, response))
+                        )
+                )
+                .flatMap(this::deAnonymizedInternalId);
+    }
+
+    private Mono<ProfileResponse> deAnonymizedInternalId(ProfileResponse response) {
+        return Flux.fromIterable(response.getDelegateMandates())
+                .flatMap(mandate -> dataVaultClient.deAnonymized(mandate.getDelegatorInternalId())
+                        .map(taxId -> {
+                            mandate.setTaxId(taxId);
+                            return mandate;
+                        }))
+                .collectList()
+                .map(updatedMandates -> {
+                    response.setDelegateMandates(updatedMandates);
+                    return response;
+                })
+                .flatMap(profileResponse ->
+                        Flux.fromIterable(profileResponse.getDelegatorMandates())
```

Comment:
wrappiamo quello che c'è nel flatMap in un metodo

Reply:
come richiesto, abbiamo wrappato il contenuto del metodo flatMap in un altro metodo privato